engine/access/ingestion/engine.go

```diff
@@ -268,6 +271,14 @@ func (e *Engine) handleExecutionReceipt(originID flow.Identifier, r *flow.Execut
 		return fmt.Errorf("failed to store execution receipt: %w", err)
 	}
 
+	// TODO - handle possibly conflicting execution results in a proper way
+	// for example, for unsealed blocks any ER is valid, but for sealed blocks
+	// there should match the seal data.
+	err = e.executionResults.ForceIndex(r.ExecutionResult.BlockID, r.ExecutionResult.ID())
```

Comment:
It does store it.
This just uses saved last received one instead of the first one, possibly solving problem that if we get new Receipt its result won't get updated. This way its consistent

Reply:
Is this index used for looking up _one_ result by block ID, or _all_ results by block ID?

If whenever we receive a receipt, we will store the receipt, index the receipt by the block ID, store the result, and index the result by the block ID. Then, we can still get all results by the block ID without forcing updating the index. No?

Reply:
One.
We store multiple execution receipts (they have their own ID) and multiple execution results, but only map blockID -> execution result one to one.

Reply:
Could you remind me why access node stores a one to one mapping from blockID to execution result?

Do we need to ensure this operation is concurrent safe? Like, if the access node receives two receipts for the same block with different result IDs concurrently, does it matter which one get indexed in the end?

Reply:
I think I get it:

Without this fix, if an AN receives a receipt with a bad result, and later receives the good result. Then when AN is queried a result by a block ID, it will return the bad result.

With this fix, in the above case, then AN will return the good result.

With the future proper fix, then AN never change the returned result for sealed blocks.

Reply:
> Could you remind me why access node stores a one to one mapping from blockID to execution result?

Good question, but I have no idea, it "was like this already", but it's useful now.

> Do we need to ensure this operation is concurrent safe? Like, if the access node receives two receipts for the same block with different result IDs concurrently, does it matter which one get indexed in the end?

In general - yes, I would assume in such case we would need to only save the one from sealed block, but that's for later.
For now we I just added this to make sure things are consistent - ie new execution receipt saves new execution results, thats it

Reply:
> I think I get it:

Yes! Exactly what you wrote. Sorry if I didn't make it clear